From 16a92d8fa7081e7e706f65e59cb49aabbeba030c Mon Sep 17 00:00:00 2001
From: BOREDDY MAHIDHAR <quic_bmahidha@quicinc.com>
Date: Thu, 11 Jan 2024 13:10:18 +0530
Subject: [PATCH 3/5] sf: Add check for unknown dataspace

If dataspace returned from gralloc is Unknown, then drawingstate
data space need to be set.

Change-Id: Id7e1ccef770c465e3af5f7aa8c310e433a423228
CRs-Fixed: 3713320
---
 services/surfaceflinger/Layer.cpp | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/services/surfaceflinger/Layer.cpp b/services/surfaceflinger/Layer.cpp
index 0be6bc0c43..5b09a3852e 100644
--- a/services/surfaceflinger/Layer.cpp
+++ b/services/surfaceflinger/Layer.cpp
@@ -14,6 +14,12 @@
  * limitations under the License.
  */
 
+/* Changes from Qualcomm Innovation Center are provided under the following license:
+ *
+ * Copyright (c) 2023-2024 Qualcomm Innovation Center, Inc. All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause-Clear
+ */
+
 // TODO(b/129481165): remove the #pragma below and fix conversion issues
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wconversion"
@@ -3684,7 +3690,10 @@ void Layer::gatherBufferInfo() {
             ATRACE_NAME("getDataspace");
             err = mapper.getDataspace(mBufferInfo.mBuffer->getBuffer()->handle, &dataspace);
         }
-        if (err != OK || dataspace != mBufferInfo.mDataspace) {
+        if ((err != OK || dataspace != mBufferInfo.mDataspace)
+            /* QTI_BEGIN */
+            && dataspace != ui::Dataspace::UNKNOWN) {
+            /* QTI_END */
             {
                 ATRACE_NAME("setDataspace");
                 err = mapper.setDataspace(mBufferInfo.mBuffer->getBuffer()->handle,
-- 
2.45.2

